services/user-management/src/main/java/com/foodie/user/repositories/UserRoleRepository.java

```diff
@@ -0,0 +1,32 @@
+package com.foodie.user.repositories;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import com.foodie.FoodieBaseRepository;
+import com.foodie.user.model.UserRole;
+
+import jakarta.transaction.Transactional;
+
+@Repository
+public interface UserRoleRepository extends FoodieBaseRepository<UserRole, UUID> {
+
+    @Modifying
+    @Transactional
+    @Query("update UserRole ur set ur.deletedAt = CURRENT_TIMESTAMP where ur.userId = :userId")
+    void deleteAllByUserId(@Param("userId") UUID id);
+
+    @Query("SELECT ur FROM UserRole ur JOIN ur.userId u WHERE u.id = :userId AND ur.deletedAt IS NULL")
```

Comment:
If your entity classes are created correctly then you wouldn't need to have this queries here. I imagine you would be able to accomplish the same goals using magic methods

Reply:
Due to the onetomany mapping. This query is needed